store.go

```diff
@@ -2051,6 +2067,18 @@ func dedupeNames(names []string) []string {
 }
 
 func (s *store) SetNames(id string, names []string) error {
+	return s.updateNames(id, names, setNames)
+}
+
+func (s *store) AddNames(id string, names []string) error {
```

Comment:
I’m not sure whether to go there — others, feel free to say this is not the time.

If we are adding a new top-level API, do we _have_ to follow the generic `SetNames` model, where the code needs to lock all three stores and guess which one the user meant? I’d expect the user to _always_ know whether the goal is to set a name of a container/layer/image. So we could have `Add{Container,Image,Layer}Names`, that would actually be more efficient than the API provided now, and the tedious `switch` statements in `store.updateNames` would not be necessary.